kythe/java/com/google/devtools/kythe/platform/shared/filesystem/ByteBufferByteChannel.java

```diff
@@ -23,13 +23,20 @@
 import java.nio.channels.ClosedChannelException;
 import java.nio.channels.NonWritableChannelException;
 import java.nio.channels.SeekableByteChannel;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
 
-/** Read-only {@link SeekableByteChannel} backed by byte[] data. */
+/** Read-only {@link SeekableByteChannel} lazily backed by byte[] data. */
 class ByteBufferByteChannel implements SeekableByteChannel {
-  private byte[] buffer;
+  private Future<byte[]> buffer;
   private long position;
 
   public ByteBufferByteChannel(byte[] data) {
```

Comment:
I don't think this constructor is used. I'd remove it, and the reference to CompetableFuture.